include/circt/Dialect/FIRRTL/Passes.h

```diff
@@ -117,8 +117,18 @@ std::unique_ptr<mlir::Pass> createPrintNLATablePass();
 std::unique_ptr<mlir::Pass>
 createBlackBoxReaderPass(std::optional<mlir::StringRef> inputPrefix = {});
 
+enum class CompanionMode {
+  // Lower companions to SystemVerilog binds.
+  Bind,
+  // Lower companions to explicit instances. Used when assertions or other
+  // debugging constructs from the companion are to be included in the design.
+  Instantiate,
+  // Drop companion modules, eliminating them from the design.
+  Drop,
+};
+
 std::unique_ptr<mlir::Pass>
-createGrandCentralPass(bool instantiateCompanionOnly = false);
+createGrandCentralPass(std::optional<CompanionMode> companionMode = {});
```

Comment:
This formulation as an optional companion mode would be better as an explicit companion mode. I'd also say drop the default mode so that there isn't duplication of defaults in `firtool` and the pass.